src/main/java/seedu/address/model/student/Class.java

```diff
@@ -332,11 +332,23 @@ public boolean equals(Object other) {
 
     /**
      * Returns 1 is this {@code Class} starts before the given {@code aclass}.
-     * {@code Class} and {@code aclass} must be non-null;
+     * {@code Class} and {@code aclass} must be non-null and on the same day;
      */
     public int compareToByStartTime(Class aclass) {
-        requireAllNonNull(this.startTime, aclass.startTime);
+        requireAllNonNull(this.date, this.startTime, aclass.date, aclass.startTime);
+        assert this.date.equals(aclass.date);
         return this.startTime.compareTo(aclass.startTime);
     }
 
+    /**
+     * Returns 1 is the {@code Class} starts before the given {@code aclass}.
```

Comment:
Do you mean `Returns 1 if ...`?

Reply:
Yes and here it should be `returns -1 if ...`. Thanks for bringing it up!